Frontend_Web_Development_React_Angular_Vue/React/Week 1/REACT-1.6 Create and initialize a React App.md

```diff
@@ -0,0 +1,131 @@
+<a href="https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-18-04">How To Install Node.js on Ubuntu 18.04.</a>
+---
+---
+
+## Creating a New Project Using  React App
+* In this first step, We wil create a new application using the npm package manager to run a remote script. The script will copy the necessary files into a new directory and install all dependencies.
+
+ * When we installed Node, we also installed a package managing application called **npm**. 
+---
+## What Is Npm And How It work 
+
+* NPM – or "Node Package Manager" – is the default package manager for JavaScript's runtime Node.js.
+
+* Npm also includes a tool called **npx**, which will run executable packages. What that means if we will run the Create React App code without first downloading the project.
+---
+## Differntiate between Npm nd Npx
+* NPM - Manages packages but doesn't make life easy executing any.
+NPX - A tool for executing Node packages.
+NPX comes bundled with NPM version 5.2+
+
+* NPM by itself does not simply run any package. it doesn't run any package in a matter of fact. If you want to run a package using NPM, you must specify that package in your package.json file.
+---
+ 
+## Creating A React App
+* The executable package will run the installation of create-react-app into the directory that you specify.
+* It will start by making a new project in a directory, which in this tutorial will be called as  **girlscript-Winter-Of-contribution** .
+*  Again, this directory does not need to exist beforehand; the executable package will create it for you. The script will also run **npm install** inside the project directory, which will download any additional dependencies.
+---
+* Step-1:- To start, create a new application using Create React App on your local machine. In a **terminal**, run the command to build an application called:**girlscript-Winter-Of-contribution-app**
+<p>&nbsp;</p>
+
+### Command to create react App:
+ ```jsx
+ npx create-react-app girlscript-Winter-Of-contribution-app
+
+```
+ 
+## How This Command Works
+* The **npx** command will run a Node package without downloading it to your machine.
+*  The **create-react-app** script will install all of the dependencies and will build a base project in the **girlscript-Winter-Of-contribution-app** directory.
+
+* The code will download the dependencies and will create a base project. It may take a few minutes to finish. When it is complete, you will receive a **success** message. Your version may be slightly different if you use **yarn** instead of **npm**
+*  As Yarn is a package manager for our code that helps to manage project dependencies like NPM.
```

Comment:
Add everything about yarn in the same bullet point. Also, there's a missing space between 'the' and 'yarn add command'. Kindly preview your file as to how it looks before making any update.

Reply:
sir, I have changed it plz check it

Reply:
Looks good.